### PR TITLE
Feature/date picker

### PR DESCRIPTION
**Pull request by Unsaved2**
This pull request adds a DatePickerDialog to `AddTravel`, `AddActivity`, `EditTravel` and `EditActivity`. 

Due to code duplication, I created a new utility class `DateTimeUtils` to handle the date and time and grouped the DatePickerDialog, TimePickerDialog and a function that transforms a string into a Timestamp into this class.

The different screens now have a button to select a date and to be consistent there is also a button for the time. Like this, it improves the user experience.

The tests have been modified accordingly.

### CODE REVIEW

File: app/src/main/java/com/github/se/travelpouch/ui/dashboard/EditActivity.kt [hunk 1]
```diff
@@ -119,9 +119,25 @@ fun EditActivity(navigationActions: NavigationActions, activityViewModel: Activi
...
                   visualTransformation = DateVisualTransformation(),
                   keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
-                  modifier = Modifier.testTag("dateField"))
+                  modifier = Modifier.fillMaxWidth().testTag("dateField"),
+                  trailingIcon = {
+                    IconButton(
+                        onClick = {
+                          dateTimeUtils.showDatePicker(context) { selectedDate ->
+                            date =
+                                selectedDate.replace(
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/ui/dashboard/EditActivity.kt:131]: As was explained to me in the DMs, this is weird and seems redundent but there's a good reason for this inelegant and counter intuitive code. Visual Transformer messes with the format, it's out of scope to refactor it at this stage so this band aid will suffice for now.

**Review by RemIsMyWaifuu — APPROVED** [review]
I have no particular nitpick or comment to say. 
This is simply a great piece of code and work, exactly as described, covered by enough test for +80% line coverage.
Good initiative to compress the time/date handling into a cohesive class.

File: app/src/main/java/com/github/se/travelpouch/ui/dashboard/EditActivity.kt (hunk 1 above)
Comment (Unsaved2) [reply app/src/main/java/com/github/se/travelpouch/ui/dashboard/EditActivity.kt:131]: Yes like we say in the DM, it is not really elegant to do this like that. I will check if I can do it better.

Comment (Unsaved2) [reply app/src/main/java/com/github/se/travelpouch/ui/dashboard/EditActivity.kt:131]: I haven't found a way to make it more elegant without adding more complexity. We will keep it like this.

**Pull request merged**